cylc/flow/simulation.py

```diff
@@ -37,6 +41,56 @@
     from cylc.flow.task_proxy import TaskProxy
 
 
+class ModeSettings:
```

Comment:
Neat, but probably gives the same result as a `NamedTuple` or a `@dataclass`?

Reply:
I need to have a look at whether we have the memory reserving properties of `__slots__`, but I bet that you are right!